### PR TITLE
Use structs to pass data to scheduler prints

### PR DESCRIPTION
**Pull request by dmcilvaney**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [ ] The toolchain has been rebuilt successfully (or no changes were made to it)
- [ ] The toolchain/worker package manifests are up-to-date
- [ ] Any updated packages successfully build (or no packages were changed)
- [ ] Packages depending on static components modified in this PR (Golang, `*-static` subpackages, etc.) have had their `Release` tag incremented.
- [ ] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [ ] All package sources are available
- [ ] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [ ] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/data/licenses.json`, `./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [ ] All source files have up-to-date hashes in the `*.signatures.json` files
- [ ] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [ ] Documentation has been updated to match any changes to the build system
- [ ] If you are adding/removing a .spec file that has multiple-versions supported, please add [@microsoft/cbl-mariner-multi-package-reviewers](https://github.com/orgs/microsoft/teams/cbl-mariner-multi-package-reviewers) team as reviewer [(Eg. golang has 2 versions 1.18, 1.21+)](https://github.com/microsoft/azurelinux/tree/2.0/SPECS/golang)
- [ ] Ready to merge

---

###### Summary 

In preparation for adding more data (licensing checks) to the build summary, use structs to pass build data around. The function signatures were getting unmanageable.

###### Change Log  

- Pass data inside `PrintBuildResult()` via new structs: `srpmBuildDataContainer` and `srpmTestDataContainer`.

###### Does this affect the toolchain?  

**NO**

###### Associated issues  

- #xxxx

###### Links to CVEs  
- https://nvd.nist.gov/vuln/detail/CVE-YYYY-XXXX

###### Test Methodology

- Pipeline build id: xxxx

### CODE REVIEW

File: toolkit/tools/scheduler/schedulerutils/printresults.go [hunk 1]
```diff
@@ -224,16 +241,18 @@ func buildResultsSetToNodesSet(statesSet map[string]*BuildResult) (result map[st
 	return
 }
 
-func getSRPMsState(pkgGraph *pkggraph.PkgGraph, buildState *GraphBuildState) (failedSRPMs map[string]*BuildResult, prebuiltSRPMs, prebuiltDeltaSRPMs, builtSRPMs map[string]bool, blockedSRPMs map[string]*pkggraph.PkgNode) {
-	failedSRPMs = make(map[string]*BuildResult)
-	prebuiltSRPMs = make(map[string]bool)
-	prebuiltDeltaSRPMs = make(map[string]bool)
-	builtSRPMs = make(map[string]bool)
-	blockedSRPMs = make(map[string]*pkggraph.PkgNode)
+func getSRPMsState(pkgGraph *pkggraph.PkgGraph, buildState *GraphBuildState) (srpmData srpmBuildDataContainer) {
```
Comment (mfrw) [inline toolkit/tools/scheduler/schedulerutils/printresults.go:244]: How about returning a pointer ?
```suggestion
func getSRPMsState(pkgGraph *pkggraph.PkgGraph, buildState *GraphBuildState) (srpmData *srpmBuildDataContainer) {
```

or maybe drop the explit named return all together :) [just an opinion]

```go
func getSRPMsState(pkgGraph *pkggraph.PkgGraph, buildState *GraphBuildState)  *srpmBuildDataContainer {
```

File: toolkit/tools/scheduler/schedulerutils/printresults.go [hunk 2]
```diff
@@ -242,51 +261,53 @@ func getSRPMsState(pkgGraph *pkggraph.PkgGraph, buildState *GraphBuildState) (fa
...
 
-		_, found := failedSRPMs[node.SrpmPath]
+		_, found := srpmData.failedSRPMs[node.SrpmPath]
 		if !found {
-			blockedSRPMs[node.SrpmPath] = node
+			srpmData.blockedSRPMs[node.SrpmPath] = node
 		}
 	}
 
 	return
```
Comment (mfrw) [inline toolkit/tools/scheduler/schedulerutils/printresults.go:280]:
```suggestion
	return &srpmData
```

File: toolkit/tools/scheduler/schedulerutils/printresults.go [hunk 3]
```diff
@@ -242,51 +261,53 @@ func getSRPMsState(pkgGraph *pkggraph.PkgGraph, buildState *GraphBuildState) (fa
...
 }
 
-func getSRPMsTestsState(pkgGraph *pkggraph.PkgGraph, buildState *GraphBuildState) (failedSRPMsTests map[string]*BuildResult, skippedSRPMsTests, passedSRPMsTests map[string]bool, blockedSRPMsTests map[string]*pkggraph.PkgNode) {
-	failedSRPMsTests = make(map[string]*BuildResult)
-	skippedSRPMsTests = make(map[string]bool)
-	passedSRPMsTests = make(map[string]bool)
-	blockedSRPMsTests = make(map[string]*pkggraph.PkgNode)
+func getSRPMsTestsState(pkgGraph *pkggraph.PkgGraph, buildState *GraphBuildState) (srpmTestData srpmTestDataContainer) {
```
Comment (mfrw) [inline toolkit/tools/scheduler/schedulerutils/printresults.go:283]: Similar idea as above :)

File: toolkit/tools/scheduler/schedulerutils/printresults.go (hunk 1 above)
Comment (dmcilvaney) [reply toolkit/tools/scheduler/schedulerutils/printresults.go:244]: Curious why you prefer the pointer, afaik a struct with non-primative types (lists/maps etc.) will be a lightweight copy that references the same underlying lists/maps without duplication.

File: toolkit/tools/scheduler/schedulerutils/printresults.go [hunk 4]
```diff
@@ -372,6 +393,16 @@ func printErrorInfoByCondition(condition bool, format string, arg ...any) {
 	}
 }
 
+// printWarningInfoByCondition prints warning or info level logs depending on the input condition.
+// If the condition is true, it prints a warning level log and an info level one otherwise.
+func printWarningInfoByCondition(condition bool, format string, arg ...any) {
```
Comment (neha170) [inline toolkit/tools/scheduler/schedulerutils/printresults.go:398]: Is this function being used anywhere? I couldn't find a usage

Comment (dmcilvaney) [reply toolkit/tools/scheduler/schedulerutils/printresults.go:398]: https://github.com/microsoft/azurelinux/pull/9885

Comment (cwize1) [reply toolkit/tools/scheduler/schedulerutils/printresults.go:398]: Remove it from this PR then?

**Review by cwize1 — APPROVED** [review]

File: toolkit/tools/scheduler/schedulerutils/printresults.go [hunk 5]
```diff
@@ -372,6 +393,16 @@ func printErrorInfoByCondition(condition bool, format string, arg ...any) {
...
+	if condition {
+		logger.Log.Warnf(color.YellowString(format, arg...))
+	} else {
+		logger.Log.Infof(color.GreenString(format, arg...))
+	}
+}
+
```
Comment (dmcilvaney) [inline toolkit/tools/scheduler/schedulerutils/printresults.go:405]:
```suggestion
```

**Pull request merged**